### PR TITLE
added support for proxy_protocol

### PR DESCRIPTION
**Pull request by jtlamers**

### CODE REVIEW

**Comment by tjikkun** [comment]
Ik zou hem rebasen naar upstream master

File: manifests/resource/mailhost.pp [hunk 1]
```diff
@@ -173,6 +181,61 @@
     fail('You must include the nginx base class before using any defined resources')
   }
 
+  if is_string($listen_port) {
```
Comment (tjikkun) [inline manifests/resource/mailhost.pp]: This should not be added I think

File: spec/defines/resource_mailhost_spec.rb [hunk 2]
```diff
@@ -298,6 +310,7 @@
           end
         end
 
+<<<<<<< HEAD
```
Comment (tjikkun) [inline spec/defines/resource_mailhost_spec.rb]: This should not be here

File: spec/defines/resource_mailhost_spec.rb [hunk 3]
```diff
@@ -551,6 +564,19 @@
...
+              value: 'off',
+              match: '  proxy_protocol        off;'
+            },
+            {
+              title: 'should set proxy_smtp_auth',
+              attr: 'proxy_smtp_auth',
+              value: 'off',
+              match: '  proxy_smtp_auth       off;'
+            },
+
```
Comment (tjikkun) [inline spec/defines/resource_mailhost_spec.rb]: extra newline should not be here

File: templates/mailhost/mailhost.erb [hunk 4]
```diff
@@ -37,7 +37,6 @@ server {
   <%- end -%>
 <%- end -%>
 <%= scope.function_template(["nginx/mailhost/mailhost_common.erb"]) -%>
-
```
Comment (tjikkun) [inline templates/mailhost/mailhost.erb]: Why did you remove this?

**Review by tjikkun — CHANGES_REQUESTED** [review]

File: manifests/resource/mailhost.pp (hunk 1 above)
Comment (jtlamers) [reply manifests/resource/mailhost.pp]: adjusted

File: spec/defines/resource_mailhost_spec.rb (hunk 2 above)
Comment (jtlamers) [reply spec/defines/resource_mailhost_spec.rb]: adjusted

File: spec/defines/resource_mailhost_spec.rb (hunk 3 above)
Comment (jtlamers) [reply spec/defines/resource_mailhost_spec.rb]: removed newline

File: templates/mailhost/mailhost.erb (hunk 4 above)
Comment (jtlamers) [reply templates/mailhost/mailhost.erb]: readded newline

**Review by tjikkun — APPROVED** [review]

**Pull request merged**